src/site/_collections/authors.js

```diff
@@ -71,56 +71,68 @@ module.exports = (collections) => {
-        : [];
-
-      // If the author doesn't have any posts, use their Twitter profile.
-      if (author.elements.length === 0) {
-        if (!author.twitter) {
-          // If we're screenshot testing just ignore it.
-          if (process.env.PERCY) {
-            return;
-          }
-
-          // posts.length might be empty if we're generating partials so only
-          // log a warning if it has a value and we're doing a regular build.
-          // Note this is checking for _all_ posts and not just the posts
-          // by this specific author.
-          if (posts.length) {
-            console.warn(
-              `author ${
-                author.title
-              } has no posts and no social: ${JSON.stringify(author)}\n`,
-            );
-          }
-        } else {
-          author.href = `https://twitter.com/${author.twitter}`;
+  Object.keys(contributors).forEach((key) => {
+    const author = contributors[key];
+    author.key = key;
+
+    // Generate the contributor's name out of valid given/family parts. This
+    // allows our authors to just have a single name.
+    const parts = [author.name.given, author.name.family].filter(
+      (s) => s && s.length,
+    );
+    author.title = parts.join(' ');
+    author.href = `/authors/${key}/`;
+
+    author.description =
+      author.description && author.description.en
+        ? author.description.en
+        : `Our latest news, updates, and stories by ${author.title}.`;
+
+    // This updates the shared contributors object with meta information and is safe to be called multiple times.
+    author.url = path.join('/en', author.href);
+    author.data = {
+      title: author.title,
+      subhead: author.description,
+    };
+
+    author.elements = authorsPosts.has(key) ? authorsPosts.get(key) : [];
+
+    // If the author doesn't have any posts, use their Twitter profile.
+    if (author.elements.length === 0) {
+      if (!author.twitter) {
+        // If we're screenshot testing just ignore it.
+        if (process.env.PERCY) {
+          return;
         }
-      }
 
-      const authorsImage = findAuthorsImage(author.key);
-      if (authorsImage) {
-        author.data.hero = authorsImage;
-        author.data.alt = author.title;
+        // posts.length might be empty if we're generating partials so only
+        // log a warning if it has a value and we're doing a regular build.
+        // Note this is checking for _all_ posts and not just the posts
+        // by this specific author.
+        if (posts.length) {
+          console.warn(
+            `author ${
+              author.title
+            } has no posts and no social: ${JSON.stringify(author)}\n`,
+          );
+        }
+      } else {
+        author.href = `https://twitter.com/${author.twitter}`;
       }
+    }
 
-      if (process.env.PERCY) {
-        author.elements = author.elements.slice(-6);
-      }
+    const authorsImage = findAuthorsImage(key);
```

Comment:
I wonder how slow this function is since it needs to do 3 disk reads per author... I don't think we have any 3x photos at the moment and we encourage everyone to add a 2x photo. If there are any existing cases without a 2x photo maybe we should just duplicate their 1x, rename it 2x, and always use the 2x version for all of our authors?